src/Npgsql/NpgsqlStream.cs

```diff
@@ -0,0 +1,329 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Npgsql
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    public abstract class NpgsqlStream : Stream
+    {
+        #region Constructor
+        /// <inheritdoc />
+        internal protected NpgsqlStream(bool canRead = false, bool canSeek = false, bool canWrite = false)
```

Comment:
Let's remove `protected` modifier, so no one outside of Npsql will be able to derive from the class.

Reply:
Done.